hooks/useStore.js

```diff
@@ -12,17 +12,38 @@ const useStore = create(
             id: 1,
             shown: false,
             title: 'Saved',
-            description:
-              'Your Character was saved. Check it out under "Characters"',
+            description: 'Your Character was saved.',
+          },
+          {
+            id: 2,
+            shown: false,
+            title: 'Updated',
+            description: 'Your Character was updated.',
+          },
+          {
+            id: 3,
+            shown: false,
+            title: 'Deleted',
+            description: 'Your Character was deleted.',
           },
         ],
 
         addCharacter: newCharacter => {
           set(state => {
             return {
               characters: [newCharacter, ...state.characters],
+              toasts: state.toasts.map(toast =>
+                toast.id === 1 ? { ...toast, shown: true } : toast
+              ),
             };
           });
+          setTimeout(() => {
+            set(state => {
+              return {
+                toasts: state.toasts.map(toast => ({ ...toast, shown: false })),
+              };
+            });
+          }, 2000);
```

Comment:
You are doing this toast logic three times with only the toast.id having different values. This screams like you should write a function for this where you generalize this behaviour. Then you can call this function simply when you need this functionality.